asn1/LedgerLedgerLotocol.asn

```diff
@@ -0,0 +1,135 @@
+LedgerLedgerLotocol
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    UInt8,
+    UInt32,
+    UInt64,
+    UInt128,
+    UInt256
+    FROM GenericTypes
+
+    Address,
+    Timestamp
+    FROM InterledgerTypes
+;
+
+FreeRequest ::= SEQUENCE {
+    -- Used to associate requests and corresponding responses
+    -- MUST be either zero and non-repeating
+    -- If requestId = 0, the server MUST not send a response
+    requestId UInt128,
+    packet InterledgerPacket
+}
+
+FreeResponse ::= SEQUENCE {
+    -- Used to associate response to earlier FreeRequest
+    requestIdRef UInt128,
+    packet InterledgerPacket
+}
+
+PaidRequest ::= SEQUENCE {
+    -- Used to associate requests and corresponding responses
+    -- MUST be either zero and non-repeating
+    -- If requestId = 0, the server MUST not send a response
+    requestId UInt128,
+    amount UInt64,
+    packet InterledgerPacket
+}
+
+PaidResponse ::= SEQUENCE {
+    -- Used to associate response to earlier PaidRequest
+    requestIdRef UInt128,
+    packet InterledgerPacket
+}
+
+PrepareRequest ::= SEQUENCE {
+    -- Used to associate later response/fulfill/reject
+    -- MUST be non-zero and non-repeating
+    requestId UInt128,
```

Comment:
I think we should separate requestId from transferId. Otherwise we can get confusing scenarios where many requests/responses are sent with the same ID, i.e. if someone tries to fulfill twice or fails to fulfill and then rejects